src/main/java/org/opentripplanner/updater/transit/ticket/parser/ConstraintsParser.java

```diff
@@ -0,0 +1,56 @@
+package org.opentripplanner.updater.transit.ticket.parser;
+
+import org.opentripplanner.model.Route;
+import org.opentripplanner.model.Stop;
+import org.opentripplanner.pricing.transit.ticket.TransitTicket;
+import org.opentripplanner.pricing.transit.ticket.pattern.RoutePattern;
+import org.opentripplanner.pricing.transit.ticket.pattern.StopPattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class ConstraintsParser {
```

Comment:
I have a feeling that it is the only parser that is not boilerplate. Making own json parsers could have fatal consequences, please check if you could just use default parsers for other classes. It may require changing format a bit, but it would be much cleaner.